File: java/src/observer/weatherobservable/StatisticsDisplay.java
package observer.weatherobservable;

import java.util.*;

public class StatisticsDisplay implements Observer, DisplayElement {
	Observable observable;
	private float maxTemp = 0.0f;
	private float minTemp = 200;
	private float tempSum= 0.0f;
	private int numReadings;

	public StatisticsDisplay(Observable observable) {
		this.observable = observable;
		observable.addObserver(this);
	}

	public void update(Observable obs, Object arg) {
		if(obs instanceof WeatherData) {
			WeatherData weatherData = (WeatherData)obs;
			float temp = weatherData.getTemperature();
			tempSum += temp;
			numReadings++;

			if (temp > maxTemp) {
				maxTemp = temp;
			}

			if (temp < minTemp) {
				minTemp = temp;
			}

		display();

		}
	}

	public void display() {
		System.out.println("Avg/Max/Min temperature = " + (tempSum / numReadings)
			+ "/" + maxTemp + "/" + minTemp);
	}
}

File: php/observer/StatisticsDisplay.php
<?
include_once("WeatherData.php");
include_once("Observer.php");
include_once("DisplayElement.php");

class StatisticsDisplay implements Observer, DisplayElement {
    private $maxTemp;
    private $minTemp;
    private $tempSum;
    private $numReadings;
    private $weatherData;

    public function __construct(WeatherData $weatherData) {
        $this->maxTemp = 0.0;
        $this->minTemp = 200;
        $this->tempSum = 0.0;
        $this->numReadings = 0;
        $this->weatherData = $weatherData;
        $this->weatherData->registerObserver($this);
    }

    public function update($temp, $humidity, $pressure) {
        $this->tempSum += $temp;
        $this->numReadings += 1;

        if($temp > $this->maxTemp) {
            $this->maxTemp = $temp;
        }
        if($temp < $this->minTemp) {
            $this->minTemp = $temp;
        }

        $this->display();
    }

    public function display() {
        echo "Avg/Max/Min temperature = $this->tempSum/$this->maxTemp/$this->minTemp";
    }
}
?>